### DISCUSSION TITLE
Not all associated pull requests showing up in "Generate Releasenotes V3"

### DISCUSSION THREAD

**Discussion by markbangert** [rfennell/AzurePipelines#973] [Q&A] [answered]
I am working with a staged deployment with three stages dev, qual, and prod. When deploying to qual our pipeline generates release notes based on a diff to the currently deployed prod release. Everything works fine until I update qual _without_ updating prod. In this scenario I see all associated commits in the release notes but only the pull request that happened between the update of qual and the current production deployment. I am missing the earlier pull requests. I want to lay out an example to make things clear:

Let's say release 100 is live on prod. Now I deploy release 120 to qual and automatically generate release notes. These release notes will be fine and include all associated commits and all associated pull requests. If I deploy release 140 at some later point without promoting release 120 to prod first the release notes are corrupted: While I see all commits, only pull requests happening between releases 120-140 show up in the release notes. Pull requests happening between releases 100-120 are missing. How is this possible?

Here is my code:
```
# Release notes for {{releaseDetails.releaseDefinition.name}}    
**Release Number**: {{releaseDetails.name}}
**Build Number**: {{buildDetails.id}}
**Compared Release Number**: {{compareReleaseDetails.name}}    

# Associated Pull Requests
{{#forEach pullRequests}}
	{{#if this.labels}}
		{{#contains (pluck this.labels 'name') '$(FilterTag)'}}		
* [{{this.title}}](https://dev.azure.com/basf4dev/NIRS_trinamiX/_git/{{this.repository.name}}/pullrequest/{{this.pullRequestId}})
		{{/contains}}
	{{/if}}
{{/forEach}}

# Associated Commits
{{#forEach commits}}
* **{{this.id}}** by: {{this.author.displayName}}  
   -  Commit message: {{this.message}}
{{/forEach}}
```
Note that we are using a $(FilterTag) to filter out irrelevant pull requests.

**Comment by rfennell** [comment, marked as answer]
What parameters are you using on the task?

If you use the `checkstage=true` it should do the compare against the stage it is running in, which I think is what you are after

**Reply by markbangert** [reply]
Argh... it's in the docs, I've read the docs. But somehow I still needed this pointer in the right direction. Just activated "checkstage=true" as you suggested and now the release notes are consistent.

![image](https://user-images.githubusercontent.com/10157181/110900432-0364a480-8303-11eb-9e05-71a9b1f715a6.png)

Thank you so much for your time!
